nixos/modules/services/system/nohang.nix

```diff
@@ -0,0 +1,105 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.nohang;
+
+  inherit (lib)
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ Dev380 ];
+  };
+
+  options.services.nohang = {
+    enable = mkEnableOption "nohang, a daemon that keeps system responsiveness when Linux is out of memory";
+
+    package = mkPackageOption pkgs "nohang" { };
+
+    config = mkOption {
```

Comment:
I had to do a double take on the name of this field since it looks somewhat similar to `settings`, which is something else entirely. Maybe `configPath` sounds better?

Reply:
Should be changed in [04a654d](https://github.com/NixOS/nixpkgs/pull/409254/commits/04a654d3f87b4bba320f7fc861f3539087c7ced1)